python/ray/_private/runtime_env/agent/main.py

```diff
@@ -0,0 +1,197 @@
+        type=str,
+        default=None,
+        help="Specify the path of the resource directory used by runtime_env.",
+    )
+
+    parser.add_argument(
+        "--logging-level",
+        required=False,
+        type=lambda s: logging.getLevelName(s.upper()),
+        default=ray_constants.LOGGER_LEVEL,
+        choices=ray_constants.LOGGER_LEVEL_CHOICES,
+        help=ray_constants.LOGGER_LEVEL_HELP,
+    )
+    parser.add_argument(
+        "--logging-format",
+        required=False,
+        type=str,
+        default=ray_constants.LOGGER_FORMAT,
+        help=ray_constants.LOGGER_FORMAT_HELP,
+    )
+    parser.add_argument(
+        "--logging-filename",
+        required=False,
+        type=str,
+        default=runtime_env_consts.RUNTIME_ENV_AGENT_LOG_FILENAME,
+        help="Specify the name of log file, "
+        'log to stdout if set empty, default is "{}".'.format(
+            runtime_env_consts.RUNTIME_ENV_AGENT_LOG_FILENAME
+        ),
+    )
+    parser.add_argument(
+        "--logging-rotate-bytes",
+        required=False,
+        type=int,
+        default=ray_constants.LOGGING_ROTATE_BYTES,
+        help="Specify the max bytes for rotating "
+        "log file, default is {} bytes.".format(ray_constants.LOGGING_ROTATE_BYTES),
+    )
+    parser.add_argument(
+        "--logging-rotate-backup-count",
+        required=False,
+        type=int,
+        default=ray_constants.LOGGING_ROTATE_BACKUP_COUNT,
+        help="Specify the backup count of rotated log file, default is {}.".format(
+            ray_constants.LOGGING_ROTATE_BACKUP_COUNT
+        ),
+    )
+    parser.add_argument(
+        "--log-dir",
+        required=True,
+        type=str,
+        default=None,
+        help="Specify the path of log directory.",
+    )
+    parser.add_argument(
+        "--temp-dir",
+        required=True,
+        type=str,
+        default=None,
+        help="Specify the path of the temporary directory use by Ray process.",
+    )
+
+    args = parser.parse_args()
+
+    logging_params = dict(
+        logging_level=args.logging_level,
+        logging_format=args.logging_format,
+        log_dir=args.log_dir,
+        filename=args.logging_filename,
+        max_bytes=args.logging_rotate_bytes,
+        backup_count=args.logging_rotate_backup_count,
+    )
+
+    # Setup stdout/stderr redirect files
+    out_file, err_file = open_capture_files(args.log_dir)
+    configure_log_file(out_file, err_file)
+
+    agent = RuntimeEnvAgent(
+        runtime_env_dir=args.runtime_env_dir,
+        logging_params=logging_params,
+        gcs_address=args.gcs_address,
+        temp_dir=args.temp_dir,
+        address=args.node_ip_address,
+        runtime_env_agent_port=args.runtime_env_agent_port,
+    )
+
+    # POST /get_or_create_runtime_env
+    # body is serialzied protobuf GetOrCreateRuntimeEnvRequest
+    # reply is serialzied protobuf GetOrCreateRuntimeEnvReply
+    async def get_or_create_runtime_env(request):
```

Comment:
Can you add types & returns to function definitions?

Reply:
done